### DISCUSSION TITLE
Experiment with advanced/alternative compilation options

### DISCUSSION THREAD

**Discussion by trivikr** [aws/aws-sdk-js-v3#4041] [General]
### Is your feature request related to a problem? Please describe.
Right now, the distributions are emitted by TypeScript.
There is no advanced compilation or optimization done on them.

### Describe the solution you'd like
Experiment if any of the following helps:
* Using [tsickle](https://github.com/angular/tsickle) and [Advanced Compilation](https://developers.google.com/closure/compiler/docs/api-tutorial3) of Closure Compiler.
* Using [babel](https://babeljs.io/) for transpiling.
* Using [swc](https://swc.rs/) for transpiling.

### Describe alternatives you've considered
Keep the transpilation process simple by only using TypeScript.

### Additional context
* This suggestion came up during review of blog post on install size reduction.
* Tweet for reference: https://twitter.com/trivikram/status/1446596564969930753
* Blog post on replacing [Babel&TSC with SWC](https://engineering.telia.no/blog/we-exchanged-babel-tsc-for-swc-and-it-was-worth-it)
* TypeScript documentation: [Using Babel with TypeScript](https://www.typescriptlang.org/docs/handbook/babel-with-typescript.html#babel-for-transpiling-tsc-for-types)
* Wallaby.js: [Optimizing TypeScript](https://wallabyjs.com/blog/optimizing-typescript.html)
* TheSoftwareHouse: [How to speed up your TypeScript project?](https://tsh.io/blog/how-to-speed-up-your-typescript-project/)

**Comment by AllanZhengYP** [comment]
SWC scores one more point: [Next.js 12 uses SWC for faster compiling](https://nextjs.org/blog/next-12#faster-builds-and-fast-refresh-with-rust-compiler)

**Comment by undrash** [comment]
SWC support would be very appreciated by those of us who manage our packages with Nx, since [Nx supports enabling SWC](https://nx.dev/getting-started/nx-and-typescript#use-swc-as-the-compiler).

**Comment by ajredniwja** [comment]
Converting it to a discussion for more suggestions and discussion.

**Reply by dennisolien** [reply]
any suggestions, in particular, you are looking for? I would like to be able to transpile with SWC.
